src/plugins/googledrive/server/services/operation-service.js

```diff
@@ -0,0 +1,148 @@
+'use strict';
+const path = require('path');
+const process = require('process');
+const { google } = require('googleapis');
+module.exports = ({ strapi }) => ({
+  get: async (ctx) => {
+    const SERVICE_ACCOUNT_PATH = path.join(
+      process.cwd(),
+      'config',
+      'env',
+      'google_drive_service_account.json'
+    );
+    const SCOPES = ['https://www.googleapis.com/auth/drive.metadata.readonly'];
+    const auth = new google.auth.GoogleAuth({
+      keyFile: SERVICE_ACCOUNT_PATH,
+      scopes: SCOPES,
+    });
+    const drive = google.drive({ version: 'v3', auth });
+
+    if (ctx.request.query.fileId === undefined) {
+      const res = await drive.files
+        .list({
+          pageSize: 1000,
+          includeItemsFromAllDrives: true,
+          supportsAllDrives: true,
+        })
+        .then((filesList) => filesList.data)
+        .catch((err) => err);
+
+      if (res?.files?.length === 0) {
+        return 'No files found';
+      }
+      return res.files;
+    } else {
+      const driveResponse = await drive.files
+        .get({
+          fileId: ctx.request.query.fileId,
+          fields: '*',
+          supportsAllDrives: true,
+        })
+        .then((driveResponseList) => driveResponseList.data)
+        .catch((err) => err);
+      return {
+        id: driveResponse.id,
+        name: driveResponse.name,
+        trashed: driveResponse.trashed,
```

Comment:
Should we filter out trashed files when we call `drive.files.get`?  I think we only want to return files that haven't been deleted.